### PR TITLE
[kie-issues#511] Migrate from jboss-parent to apache parent pom. 

### PR DESCRIPTION
**Pull request by baldimir**
This is part of an effort from issue https://github.com/kiegroup/kie-issues/issues/511. 

- Changes root parent pom from jboss-parent to apache parent pom.xml. 
- Fixes the build by adding missing Java module names. 
- Aligns version property names with apache parent pom. 
- Adds missing version properties. 
- Adds JavaDoc plugin configuration so the JavaDoc generation is the same as before the migration. 

Sorry for some formatting changes, the formatting was wrong in those files.

### CODE REVIEW

File: drools-drl-quarkus-extension/drools-drl-quarkus/pom.xml
Comment (porcelli) [inline drools-drl-quarkus-extension/drools-drl-quarkus/pom.xml:1]: isn't anything missing here?

File: efesto/efesto-dependencies/pom.xml [hunk 1]
```diff
@@ -38,7 +38,8 @@
   </description>
 
   <properties>
-    <skip.jdepend>true</skip.jdepend>
+    <java.module.name>org.drools.efesto.dependencies</java.module.name>
+    <skip.jdepend>false</skip.jdepend>
```
Comment (porcelli) [inline efesto/efesto-dependencies/pom.xml]: why this change from true to false?

**Review by porcelli — CHANGES_REQUESTED** [review]

File: efesto/efesto-dependencies/pom.xml (hunk 1 above)
Comment (baldimir) [reply efesto/efesto-dependencies/pom.xml]: Oh my mistake, I will fix it.

File: drools-drl-quarkus-extension/drools-drl-quarkus/pom.xml
Comment (baldimir) [reply drools-drl-quarkus-extension/drools-drl-quarkus/pom.xml:1]: I will revert this file, as there are only formatting changes.

**Comment by baldimir** [comment]
@porcelli I added fixes based on your comments. Please rereview.

**Review by porcelli — APPROVED** [review]
lgtm!

**Review by jstastny-cz — APPROVED** [review]
Looks good to me.

**Review by mariofusco — APPROVED** [review]

**Pull request merged**